examples/ItemSelector3-custom-states.cc

```diff
@@ -0,0 +1,380 @@
+/*
+  Copyright (c) [2019] SUSE LLC
+  This library is free software; you can redistribute it and/or modify
+  it under the terms of the GNU Lesser General Public License as
+  published by the Free Software Foundation; either version 2.1 of the
+  License, or (at your option) version 3.0 of the License. This library
+  is distributed in the hope that it will be useful, but WITHOUT ANY
+  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+  FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+  License for more details. You should have received a copy of the GNU
+  Lesser General Public License along with this library; if not, write
+  to the Free Software Foundation, Inc., 51 Franklin Street, Fifth
+  Floor, Boston, MA 02110-1301 USA
+*/
+
+
+// Example for ItemSelector with custom status values.
+//
+// Compile with:
+//
+//     g++ -I/usr/include/yui -lyui ItemSelector3-custom-states.cc -o ItemSelector3-custom-states
+//
+// For a very simple ItemSelector example, see ItemSelector2-minimalistic.cc.
+
+#define YUILogComponent "example"
+#include "YUILog.h"
+
+#include "YUI.h"
+#include "YWidgetFactory.h"
+#include "YDialog.h"
+#include "YAlignment.h"
+#include "YButtonBox.h"
+#include "YCheckBox.h"
+#include "YFrame.h"
+#include "YIntField.h"
+#include "YItemCustomStatus.h"
+#include "YItemSelector.h"
+#include "YLabel.h"
+#include "YLayoutBox.h"
+#include "YPushButton.h"
+#include "YRadioButton.h"
+#include "YRadioButtonGroup.h"
+#include "YEvent.h"
+
+
+using std::string;
+
+
+// Config options set by askConfigValues()
+
+bool notify	  = true;
+int  visibleItems = 6;
+
+
+// The custom status values are just integers. An enum is a good and simple way
+// to use more speaking names.
+
+enum CustomStatus
+{
+    MOD_DONT_INSTALL = 0,
+    MOD_INSTALL,
+    MOD_AUTOINSTALL
+};
+
+// "Software modules" used in this example to show automatic dependency
+// handling and thus status change from the application's business logic.
+
+YDescribedItem	* modKde	= 0;
+YDescribedItem	* modXfce	= 0;
+YDescribedItem	* modX11	= 0;
+YDescribedItem	* modOffice	= 0;
+YDescribedItem	* modServer	= 0;
+YDescribedItem	* modSdk	= 0;
+
+
+// Widgets
+
+YDialog		* dialog	= 0;
+YItemSelector	* selector	= 0;
+YLabel		* resultField	= 0;
+YPushButton	* resultButton	= 0;
+YPushButton	* closeButton	= 0;
+
+
+// Function prototypes
+
+void createWidgets();
+void handleEvents();
+void handleItemClick( YItem * item );
+void handleDependencies();
+void autoSelectIfNeeded( YItem * item, bool needed );
+void updateResultField();
+void askConfigValues();
+
+
+
+int main( int argc, char **argv )
+{
+    YUILog::setLogFileName( "/tmp/libyui-examples.log" );
```

Comment:
Plus, we have been using the exact same code since at least mid-2008 in most of the other C++ UI examples.

https://github.com/libyui/libyui/blame/master/examples/ComboBox1.cc#L61
https://github.com/libyui/libyui/blame/master/examples/SelectionBox1.cc#L48
https://github.com/libyui/libyui/blame/master/examples/SelectionBox2.cc#L79
https://github.com/libyui/libyui/blame/master/examples/SelectionBox3-many-items.cc#L85
https://github.com/libyui/libyui/blame/master/examples/Table-many-items.cc#L91

Reply:
Removed the log file call. That means we no longer have a useful log file for this example; it all goes to stdout which for NCurses means it gets lost until the program exits.